server/jetstream_api.go

```diff
@@ -430,7 +477,8 @@ func (s *Server) jsTemplateCreateRequest(sub *subscription, c *client, subject,
 	if c == nil || c.acc == nil {
 		return
 	}
-	var resp JSApiStreamTemplateCreateResponse
+
+	var resp = JSApiStreamTemplateCreateResponse{ApiResponse: ApiResponse{Type: JSApiStreamTemplateCreateResponseType}}
```

Comment:
We should make a help maybe to shorten these? A bit long.

Reply:
Thought about it, but couldn't come up with something that would actually be shorter other than a ton of `NewJSApiStreamTemplateCreateResponse()` noise

Reply:
I am sure there is something, starting to look like java code.

Reply:
nothing a few factories cant solve

Reply:
best I could come up with that still avoided interfaces is either:

```
var resp = JSApiStreamTemplateCreateResponse{ApiResponse: helper(JSApiStreamTemplateCreateResponseType)} 
```

or

```
var resp = JSApiStreamTemplateCreateResponse
resp.Type = JSApiStreamTemplateCreateResponseType
```

2nd is best, but oh so easy to forget

Reply:
We have a helper when we send the responses so could runtime error if not set?

Reply:
kind of could - could do:

```go
type ApiResponse struct {
	Type  string    `json:"type"`
	Error *ApiError `json:"error,omitempty"`
}

func (r *ApiResponse) SetType(t string) {....}

type typedResponse interface {
   SetType(t string)
}

func jsonAPIResponse(c *client, subject, reply, request, response typedResponse, rtype string) {
   response.SetType(rtype)

   body, _ := json.Marshal(response)
   // etc
}

s.jsonAPIResponse(c, subject, reply, string(msg), resp, JSApiStreamTemplateCreateResponseType)
```

we can go a bit further but now it gets insane - have a map of type to type string and use reflect to find out the type of `resp` and just always set the right thing.  I doubt using reflect here is a good thing.

also avoidable with a helper like:

```go
switch resp := resp.(type) {
  case JSApiStreamTemplateCreateResponse:
     // set it
   default:
     // fail
}
```

better than reflect, and will drop the last argument - meaning its always magically set, its perhaps too magical to your tastes though

Reply:
fwiw in jsm each type that I care for has a little `SchemaType() string` on it so this all gets easier,  I can add it here too if you like and this will drop the above switch madness

Reply:
We are reaching diminishing returns, let's keep as is, I can make changes later if that important.